### PR TITLE
Add README and ci

### PR DESCRIPTION
**Pull request by shuhuiluo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,72 @@
...
+The `uni-v3-lib` by Aperture Finance is a suite of Solidity libraries that have been imported and rewritten from
+Uniswap's `v3-core` and `v3-periphery` repositories. The goal of this project is to provide external integrators with a
+set of libraries that are crucial for interaction with the Uniswap V3 protocol.
+
+## Overview
+
+This repository focuses on certain libraries from Uniswap's repositories that are considered to be most useful for
+external integrators, e.g. `SqrtPriceMath` and `TickMath`. We have modified these libraries to make them compatible with
+modern Solidity compilers (versions greater than 0.8.0). Instead of simply changing the pragma and accepting the default
+safe math behaviour, most functions have been optimized using inline assembly for less gas. Additional
```
Comment (gaohan137) [inline README.md:15]: behaviour -> Behavior

File: README.md [hunk 2]
```diff
@@ -1 +1,72 @@
...
+to the original libraries and that the optimizations have been effective. For testing against the exact same bytecode
+compiled with Solidity 0.7.6, we adopt the following approach:
+
+1. The original Uniswap libraries are exposed by test wrapper contracts in `src/test` and compiled with Hardhat using
+   the Solidity 0.7.6 compiler.
+2. We use the Forge cheatcode `vm.etch` to create these test wrappers using the bytecode from the Hardhat artifacts.
+3. Equivalence between the modified libraries and the original is verified using Foundry's fuzz testing.
+4. The effectiveness of optimizations can be assessed by tests prefixed by `testGas` in the gas snapshot.
+
+<details>
```
Comment (gaohan137) [inline README.md:41]: Let's place this comparison chart in the beginning? Basically the idea is to show the result front and center. If users of this lib is interested/hooked, they'll be motivated enough to read other details.

And let's not use the `<details>` markdown tag, it hides the comparison. We need to show it off!

Other than that, the saving is awesome! Thanks Shuhui!!

**Review by gaohan137 — COMMENTED** [review]
Thanks for the documentation!

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: Lint
```
[the remark this answers is not in the document]
Comment (shuhuiluo) [reply .github/workflows/lint.yml:1]: Separate badges.

**Review by gaohan137 — APPROVED** [review]
Fucking awesome. Let's merge!

**Pull request merged**